ocaml/xapi/xapi_vm_snapshot.ml

```diff
@@ -219,7 +219,9 @@ let checkpoint ~__context ~vm ~new_name =
 			Xapi_xenops.resume ~__context ~self:vm ~start_paused:false ~force:false;
 		end;
 		match snap with
-		| None      -> raise (Api_errors.Server_error (Api_errors.task_cancelled,[]))
+		| None  ->
+			let task_id = Context.get_task_id __context in
+			raise Api_errors.(Server_error (task_cancelled,[Ref.string_of task_id]))
```

Comment:
If snap = None then either (a) the checkpoint task was cancelled during the suspend or (b) the VM clone was cancelled. Using exn_if_cancelling at this point would catch (a) but not (b).